contrib/ros/robots/sawyer.py

```diff
@@ -33,15 +33,21 @@
 
 
 class Sawyer(Robot):
-    def __init__(self, simulated=False, control_mode='position'):
+    def __init__(self,
+                 initial_joint_pos=None,
+                 simulated=False,
+                 control_mode='position'):
         """
         :param initial_joint_pos: {str: float}
-                            {'joint_name': position_value}
+                            {'joint_name': position_value}, and also initial_joint_pos should include all os the
+                            joints that user wants to control and observe.
         """
         Robot.__init__(self)
         self._limb = intera_interface.Limb('right')
         self._gripper = intera_interface.Gripper()
-        if simulated:
+        if initial_joint_pos is not None:
+            self._initial_joint_pos = initial_joint_pos
+        elif simulated:
             self._initial_joint_pos = INITIAL_SIM_ROBOT_JOINT_POS
```

Comment:
`INITIAL_SIM_ROBOT_JOINT_POS` and `INITIAL_REAL_ROBOT_JOINT_POS` be different?

If you pepper your package with special switches for simulated vs. real, you completely lose the benefit of the sim/real abstraction layer.

The __only__ thing that `simulated=True` should change is whether the commands/state are directed to Gazebo or a real robot.